inc/field.class.php

```diff
@@ -571,11 +571,13 @@ static function showForTab($params) {
          $entities = getSonsOf(getTableForItemType('Entity'), $loc_c->fields['entities_id']);
       }
 
-      $current_entity = $item::getType() == Entity::getType()
-                           ? $item->getID()
-                           : $item->fields['entities_id'];
-      if ($item->isEntityAssign() && !in_array($current_entity, $entities)) {
-         return false;
+      if ($item->isEntityAssign()) {
+         $current_entity = $item::getType() == Entity::getType()
+                              ? $item->getID()
+                              : $item->fields['entities_id'];
```

Comment:
There is a method that already handle this case (defined by CommonDBTM and overrided by Entity).
```suggestion
         $current_entity = $item->getEntityID();
```

Reply:
Hello @cedric-anne 
If we use this, then we must test if $current_entity is not equal to -1 before doing the test "in array" (line 578) otherwise, it will return false, and the field will not be present on the form...

Reply:
`$item->getEntityID()` will not return -1 as you are in the `$item->isEntityAssign()` case.

Reply:
Hello @cedric-anne 
I thought, you were replacing all the block with this, sorry for that.